src/components/utils.ts

```diff
@@ -0,0 +1,33 @@
+class ManagedProps<T> {
+  private usedProps: (keyof T)[] = [];
+  private rawProps: T;
+
+  constructor(props: T) {
+    this.rawProps = props;
+
+    for (const key in props) {
+      // Define getters and setters dynamically
+      Object.defineProperty(this, key, {
+        get: () => {
+          this.usedProps.push(key);
```

Comment:
This wouldn't work as intended when accessing a prop with a delay (i.e. in an event), all renders until that point would have that prop in `remainingProps`.